### PR TITLE
Remove unused constants

### PR DESCRIPTION
**Pull request by practicalswift**
* Remove unused constant MEMPOOL_GD_VERSION
* Comment out unused constant REJECT_DUST

### CODE REVIEW

**Comment by achow101** [comment]
NACK `-testsafemode` is not unused. It is part of GetWarnings and will set a warning for strStatusBar, strRPC, and strGUI which will then trigger the safe mode handling of RPCs. Safe mode is still used in the RPCs to disable certain RPCs if safe mode is enabled.

**Comment by practicalswift** [comment]
@achow101 Ah, sorry. I missed that the safe mode logic in `OnRPCPreCommand` is triggered by `strRPC != nullptr` (via `GetWarnings("rpc")` :-)  Thanks for clarifying. Reverting that part of this PR.

**Comment by practicalswift** [comment]
@achow101 Updated :-)

File: src/consensus/validation.h [hunk 1]
```diff
@@ -14,7 +14,7 @@ static const unsigned char REJECT_INVALID = 0x10;
 static const unsigned char REJECT_OBSOLETE = 0x11;
 static const unsigned char REJECT_DUPLICATE = 0x12;
 static const unsigned char REJECT_NONSTANDARD = 0x40;
-static const unsigned char REJECT_DUST = 0x41;
+// static const unsigned char REJECT_DUST = 0x41;
```
Comment (achow101) [inline src/consensus/validation.h]: Since this is part of BIP 61, I think it should remain as is even though it is unused.

Comment (practicalswift) [inline src/consensus/validation.h]: @achow101 Yes, I left it commented out instead of removing it :-)

**Comment by practicalswift** [comment]
@achow101 Thanks for reviewing! Added a comment about BIP 61. Looks good?

**Comment by sipa** [comment]
utACK dce82397c1540769a8b39a9a475c366cb9438469

**Comment by achow101** [comment]
utACK dce82397c1540769a8b39a9a475c366cb9438469

**Comment by fanquake** [comment]
utACK dce8239

**Pull request merged**